ccpp/data/GFS_typedefs.F90

```diff
@@ -1729,6 +1730,14 @@ module GFS_typedefs
     real (kind=kind_phys), pointer :: toticeb(:)     => null()   !< accumulated ice precipitation in bucket (kg/m2)
     real (kind=kind_phys), pointer :: totsnwb(:)     => null()   !< accumulated snow precipitation in bucket (kg/m2)
     real (kind=kind_phys), pointer :: totgrpb(:)     => null()   !< accumulated graupel precipitation in bucket (kg/m2)
+    real (kind=kind_phys), pointer :: frzr   (:)     => null()   !< accumulated surface freezing rain (m)
+    real (kind=kind_phys), pointer :: frzrb  (:)     => null()   !< accumulated surface freezing rain in bucket (m)
+    real (kind=kind_phys), pointer :: frozr  (:)     => null()   !< accumulated surface graupel (m)
+    real (kind=kind_phys), pointer :: frozrb (:)     => null()   !< accumulated surface graupel in bucket (m)
+    real (kind=kind_phys), pointer :: tsnowp (:)     => null()   !< accumulated surface snowfall (m)
+    real (kind=kind_phys), pointer :: tsnowpb(:)     => null()   !< accumulated surface snowfall in bucket (m)
+    real (kind=kind_phys), pointer :: rhonewsn1(:)   => null()   !< precipitation ice density outside RUC LSM (kg/m3)
+    real (kind=kind_phys), pointer :: rhosnf(:)      => null()   !< precipitation ice density inside RUC LSM (kg/m3)
```

Comment:
Since you're adding rhosnf (lsm_internal_surface_frozen_precipitation_density) to the Diag DDT, you can remove the rhofr variable in the Sfcprop DDT (and any allocations/initializations/metadata) because it is replaced by rhosnf in RUC LSM.

Reply:
@ericaligo-NOAA In my branch, I repurposed the Sfcprop%rhofr variable to have the standard name of `lsm_internal_surface_frozen_precipitation_density`. In this PR branch, the Diag%rnosnf variable is used as `lsm_internal_surface_frozen_precipitation_density` instead. This is fine, but Sfcprop%rhofr is abandoned with this implementation, and should be removed.

Originally, yes, I removed Sfcprop%rhofr, but added it back in and gave it a new standard name in my branch after the discussion with Tanya.

Reply:
As long as `lsm_internal_surface_frozen_precipitation_density` exists in the metadata and the memory is allocated in fv3atm somewhere, it is fine.

Reply:
I still believe there could be an issue below because rhosnfall is populated if exticeden is false, but it's not populated if it's true. Rhosnfall is not being used currently in the LSM, but if someone decides to use it while exticeden is true, the initial negative value would be used.
      if (exticeden) then
        rhonewsn = rhonewsn_ex
        **rhosnfall = rhonewsn** <------------------------I think this should be added for consistency.
      else
        rhonewsn=min(125.,1000.0/max(8.,(17.*tanh((276.65-Tabs)*0.15))))
        rhonewgr=min(500.,rhowater/max(2.,(3.5*tanh((274.15-Tabs)*0.3333))))
        rhonewice=rhonewsn
!--- compute density of "snowfall" from weighted contribution
!                 of snow, graupel and ice fractions

        **rhosnfall** = min(500.,max(58.8,(rhonewsn*snowrat +  &
!13mar18         rhosnfall = min(500.,max(76.9,(rhonewsn*snowrat +  &
                     rhonewgr*grauprat + rhonewice*icerat + rhonewgr*curat)))

! from now on rhonewsn is the density of falling frozen precipitation
        rhonewsn=rhosnfall
      end if

Reply:
Also, my diag_tables include rhonewsn, not rhosnf
"gfs_phys",    "rhonewsn",   "rhonewsn",   "fv3_history2d",  "all",  .false.,  "none",  2

Reply:
Regarding the rhosnf question within module_sf_ruclsm, it is always only a diagnostic, but you're right that there are instances where it will be set to nonsense. 

1. Let's look at when exticeden == T.  `rhosnf` is intent(out) in the lsmruc subroutine, so it should always be set to something, but it is not. It is only initialized to  -1.e3 conditionally since it is in the `if(init .and. (lsm_cold_start) .and. iter == 1) then` block. `rhosnfall` is initialized to rhosnf (which is _uninitialized_ except for the first time step) prior to calling SFCTMP. Then, it is unmodified in SFCTMP and is used to set `rhosnf` upon return to lsmruc. So, this variable is an uninitialized diagnostic when exticeden == T. Perhaps this is OK since the externally-calculated variable (the one that is used) still has its value and the internal one is meaningless at this point.
2. Let's look at when exticeden == F. The same situation applies except that `rhosnfall` is conditionally given a value in SFCTMP. Otherwise, `rhosnf` returns nonsense. 

Critically, though, `rhosnf` potentially returning nonsense was the behavior **_before_** any of the modifications in this PR. It only ever had a value for the following conditions:

- `if(init .and. (lsm_cold_start) .and. iter == 1)` in lsmruc
- `IF((XLAND(I,J)-1.5) < 0.)` in lsmruc
- `IF(NEWSN.GT.0.)` in sfctmp

The initialization that happens to `rhosnfr` in the CCPP cap, lsm_ruc_run, is overridden by the intent(out) status of the variable in lsmruc.

I think that the concerns can be removed if `rhosnf` is declared as intent(inout) in lsmruc. That way, it should always at least have its value given at the first timestep. But, like I said, since it is only a diagnostic, it may not matter much.